anchore_engine/services/catalog/api/controllers/archives.py

```diff
@@ -177,16 +177,20 @@ def create_analysis_archive_rule(rule):
     try:
         with session_scope() as session:
             # Validate that only one system_global rule has max_images_per_account set
-            qry = session.query(ArchiveTransitionRule).filter(
-                ArchiveTransitionRule.account == ApiRequestContextProxy.namespace(),
-                ArchiveTransitionRule.system_global == True,
-                ArchiveTransitionRule.max_images_per_account != None,
-            )
-            if qry.first() is not None:
-                raise BadRequest(
-                    "A system_global Archive Transition Rule already exists with max_images_per_account set",
-                    {"existingRule": repr(qry.first())},
+            if (
+                rule.get("system_global", False)
+                and rule.get("max_images_per_account", None) is not None
+            ):
+                qry = session.query(ArchiveTransitionRule).filter(
+                    ArchiveTransitionRule.account == ApiRequestContextProxy.namespace(),
+                    ArchiveTransitionRule.system_global.is_(True),
```

Comment:
Why the change from == True to is_(True) here?  The null check on the following line I understand, but is there a postgres specific reason for this?

Reply:
@zhill i was getting warnings from my IDE saying that technically the previous check was "truthy", which I understood to mean that it was synonymous with the "IS NOT NULL" check